### DISCUSSION TITLE
Add a way to call win-wind's functions with macros

### DISCUSSION THREAD

**Discussion by tesses** [pit-ray/win-vind#96] [Tips]
I think that would be a cool idea to add a way of calling the functions with a macro

For example:
`enmap <num>gg <jump_caret_to_BOF><num><move_cursor_down>`

Or better: 
`enmap <num>gg <gg><num><j>`

I really love this project!

**Comment by pit-ray** [comment]
Surely it is hassle to have to create a mapping to call a native function, as is current win-vind.

### About first idea
The first is very cool and unique.
It is best minimalistic if written like a grouped key code without temporary mappings.
I will consider this for implementation.

### About second idea
The expression `<key>` will be likely to be a nested expression and may not be well readable.
For example.
```vim
enmap ff <shift-f><<ctrl-s>><<shift-f>>
```
An alternative is to have such feature in the command itself, as in Vim.
The `noremap` interprets predefined mappings preferentially, while the `map` gives priority to overwritten remappings.

For example.
```vim
enmap gg <move_cursor_down>
enmap <num>G gg<num>j
" <num>G  -> <move_cursor_down><num><move_cursor_down>
````

```vim
ennoremap gg <move_cursor_down>
ennoremap <num>G gg<num>j
" <num>G  ->  <jump_caret_to_BOF><num><move_cursor_down>
```

However, in this case, this is a major version change because the current `map` and `noremap` should be merged, and one must be distinguished from the other as a new command.

Please give me your opinion.
Thanks for great idea.

**Reply by tesses** [reply]
The expression `<key>` is for avoiding writing the characters when you are executing a macro.
How do you differentiate a macro with only text and a macro with commands?

Thanks for replying!

**Reply by pit-ray** [reply]
I understand that `<j>` is a shortened version of `<move_cursor_down>`.

I think the parser in `<>`  expresssion would be implemented as the following order.
1. Function name (e.g. `<move_cursor_down>`)
1. Special keywords (e.g. `<shift>`, `<space>`)
1. Commands composed of characters (e.g. `<j>`)

Thus, if `<shift-f>` is a shortened version of `<easy_click_left>`, it is confused with the macro command `<shift-f>` simply to press <kbd>shift</kbd> and <kbd>F</kbd> at the same time.

If we can resolve that issue, we can implement a shortened version.

Thanks again.

**Reply by tesses** [reply]
Oh i see that 😓.
Probably the only way is to use another character for the shortened.
I think that `{{f}}` could be a good alternative

Another good idea after the step 1) is to give the opportunity to write complex shortkeys as like as vim with `timeoutlen`.

An example of complex shortkeys:
`ciw`

Thanks

**Reply by pit-ray** [reply]
I see.

How about making `map` and `noremap` each consider native commands or not, and expressing whether or not to tell Windows about the generated commands with `{}`, etc.?


Example.
```vim
noremap j easy_click_left

" a -> move_cursor_down
noremap a j

" a -> easy_click_left
map a j

" only inside {} is actually entered.
noremap a j{This text is inserted}
```

Thanks.

**Reply by tesses** [reply]
I love it 🔝

**Comment by pit-ray** [comment]
I posted #104
